src/rt/util/typeinfo.d

```diff
@@ -36,14 +37,15 @@ if (is(T == float) || is(T == double) || is(T == real))
 
     public alias hashOf = core.internal.hash.hashOf;
 }
+
 template Floating(T)
-if (is(T == cfloat) || is(T == cdouble) || is(T == creal))
+if (is(T == c_complex_float) || is(T == c_complex_double) || is(T == c_complex_real))
 {
   pure nothrow @safe:
 
     bool equals(T f1, T f2)
     {
-        return f1 == f2;
+        return f1.re == f2.re && f1.im == f2.im;
```

Comment:
Is this change really necessary? The compiler is smart enough to generate an appropriate `opEquals` for `struct Complex(T) {T re; T im;}`

Reply:
This is an enum type, not a struct, so there is no `opEquals`, and there is no harm in inlining it explicitly.

Reply:
Actually, I'll clarify that this is only a struct type when `version=StdDdoc`.  Hence why I had to add fields to `ddoc_complex` in order to satisfy the documentation CI pipeline. :-)